pinot-broker/src/main/java/org/apache/pinot/broker/requesthandler/BaseBrokerRequestHandler.java

```diff
@@ -1168,6 +1173,56 @@ private static void handleHLLLog2mOverride(PinotQuery pinotQuery, int hllLog2mOv
     }
   }
 
+  /**
+   * Verifies that no groovy is present in the PinotQuery when disabled.
+   */
+  @VisibleForTesting
+  static void handleDisableGroovyOverride(PinotQuery pinotQuery) {
+    List<Expression> selectList = pinotQuery.getSelectList();
+    for (Expression expression : selectList) {
+      handleDisableGroovyOverride(expression);
+    }
+    List<Expression> orderByList = pinotQuery.getOrderByList();
+    if (orderByList != null) {
+      for (Expression expression : orderByList) {
+        // NOTE: Order-by is always a Function with the ordering of the Expression
+        handleDisableGroovyOverride(expression.getFunctionCall().getOperands().get(0));
+      }
+    }
+    Expression havingExpression = pinotQuery.getHavingExpression();
+    if (havingExpression != null) {
+      handleDisableGroovyOverride(havingExpression);
+    }
+    Expression filterExpression = pinotQuery.getFilterExpression();
+    if (filterExpression != null) {
+      handleDisableGroovyOverride(filterExpression);
+    }
+    List<Expression> groupByList = pinotQuery.getGroupByList();
+    if (groupByList != null) {
+      for (Expression expression : groupByList) {
+        handleDisableGroovyOverride(expression);
+      }
+    }
+  }
+
+  private static void handleDisableGroovyOverride(Expression expression) {
+    Function functionCall = expression.getFunctionCall();
+    if (functionCall == null) {
+      return;
+    }
+
+    if (functionCall.getOperator().toUpperCase().contains("GROOVY")) {
```

Comment:
The function name needs to be canonicalized. See `TransformFunctionFactory` for details

Reply:
Seems like the `canonicalize` method is for handling the multi-word transform functions like `greater_than_or_equal`. Would we still need that if we are looking for 1 term, `groovy`?

Reply:
The problem is that attacker can by-pass this check if they use `gro_ovy`, which needs to be avoided

Reply:
Makes sense. Added the canonicalization and a test case for this.